src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -717,14 +717,21 @@ private static int CalculateNumberOfExceptions(__ExceptionInfo[] excp)
             stackSize = m_stackSize;
             if (m_exceptionHeader != null && m_exceptionHeader.Length != 0)
             {
-                if (m_exceptionHeader.Length < 4)
+                // If length < 4, fail. Below pattern elides future bounds checks in method.
```

Comment:
I do not think that these point-in-time hacks are worth it for regular code non-perf critical code like this one.